server/routerlicious/packages/services-shared/src/socketIoServer.ts

```diff
@@ -190,7 +190,8 @@ class SocketIoServer implements core.IWebSocketServer {
 			if (drainTime > 0 && drainInterval > 0) {
 				// Stop receiving new connections
 				this.io.engine.use((_, res, __) => {
-					res.status(503).send("Graceful Shutdown");
+					res.writeHead(503);
+					res.end("Graceful Shutdown");
```

Comment:
So the `res` is a different type of res now?

Reply:
It is a WebSocketResponse, not sure why they claim Express middleware is fully compatible. In the code for engine.io they use writeHead and end as part of testing io.engine.use()

res.writeHead(503);
res.end("Graceful Shutdown");

Reply:
@zhangxin511 See above response from Alejandro. Also sharing the test case link where they use writeHead: https://github.com/socketio/engine.io/blob/b0721e64070c9f164fa3ef774bc785c391d7eb69/test/middlewares.js#L98